### ISSUE TITLE
How to update Agents v2.4.0 without incompatibilities

### ISSUE THREAD

**Issue by sbacelar** [JuliaDynamics/Agents.jl#470]
I have `Agents v2.4.0`on my Mac. In Windows I have a newer version. I want to update to a newer version but I keep getting incompatibilities. Is it better to create a specific environment for `Agents`?

**Comment by Libbum** [comment]
Hi @sbacelar. 2.4 is super old, so yes&mdash;it's a good idea to upgrade. The first question we'd need to know is what version of Julia are you using? I suspect it's also pretty old, since we currently support v1.5 and up. That's probably where your issues stand. If not, can you post the errors you're getting?

**Comment by sbacelar** [comment]
Hi @Libbum . I use Julia 1.6. I tried to `add Agents#master` and got the following:
```
(@v1.6) pkg> add Agents#master
    Updating git-repo `https://github.com/JuliaDynamics/Agents.jl.git`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Flux [587475ba]:
 Flux [587475ba] log:
 ├─possible versions are: 0.4.1-0.12.0 or uninstalled
 ├─restricted to versions * by an explicit requirement, leaving only versions 0.4.1-0.12.0
 ├─restricted by compatibility requirements with Requires [ae029012] to versions: 0.8.3-0.12.0 or uninstalled, leaving only versions: 0.8.3-0.12.0
 │ └─Requires [ae029012] log:
 │   ├─possible versions are: 0.5.0-1.1.3 or uninstalled
 │   └─restricted to versions 1.1.0-1 by Agents [46ada45e], leaving only versions 1.1.0-1.1.3
 │     └─Agents [46ada45e] log:
 │       ├─possible versions are: 4.2.0 or uninstalled
 │       └─Agents [46ada45e] is fixed to version 4.2.0
 ├─restricted by compatibility requirements with CUDAapi [3895d2a7] to versions: [0.4.1-0.8.3, 0.10.0-0.12.0] or uninstalled, leaving only versions: [0.8.3, 0.10.0-0.12.0]
 │ └─CUDAapi [3895d2a7] log:
 │   ├─possible versions are: 0.5.0-4.0.0 or uninstalled
 │   └─restricted by julia compatibility requirements to versions: uninstalled
 ├─restricted by compatibility requirements with CUDA [052768ef] to versions: [0.4.1-0.10.4, 0.11.2-0.12.0] or uninstalled, leaving only versions: [0.8.3, 0.10.0-0.10.4, 0.11.2-0.12.0]
 │ └─CUDA [052768ef] log:
 │   ├─possible versions are: 0.1.0-2.6.2 or uninstalled
 │   ├─restricted by compatibility requirements with DataStructures [864edb3b] to versions: 1.3.3-2.6.2 or uninstalled
 │   │ └─DataStructures [864edb3b] log:
 │   │   ├─possible versions are: 0.9.0-0.18.9 or uninstalled
 │   │   └─restricted to versions 0.18 by Agents [46ada45e], leaving only versions 0.18.0-0.18.9
 │   │     └─Agents [46ada45e] log: see above
 │   └─restricted by julia compatibility requirements to versions: [2.3.0, 2.5.0-2.6.2] or uninstalled
 ├─restricted by compatibility requirements with CodecZlib [944b1d66] to versions: 0.4.1-0.11.1 or uninstalled, leaving only versions: [0.8.3, 0.10.0-0.10.4]
 │ └─CodecZlib [944b1d66] log:
 │   ├─possible versions are: 0.4.4-0.7.0 or uninstalled
 │   ├─restricted by compatibility requirements with RDatasets [ce6b1742] to versions: 0.4.4-0.7.0
 │   │ └─RDatasets [ce6b1742] log:
 │   │   ├─possible versions are: 0.5.0-0.7.4 or uninstalled
 │   │   ├─restricted to versions * by an explicit requirement, leaving only versions 0.5.0-0.7.4
 │   │   └─restricted by compatibility requirements with DataFrames [a93c6f00] to versions: 0.6.8-0.7.4 or uninstalled, leaving only versions: 0.6.8-0.7.4
 │   │     └─DataFrames [a93c6f00] log:
 │   │       ├─possible versions are: 0.11.7-0.22.6 or uninstalled
 │   │       ├─restricted to versions [0.21-0.22, 1] by Agents [46ada45e], leaving only versions 0.21.0-0.22.6
 │   │       │ └─Agents [46ada45e] log: see above
 │   │       └─restricted by compatibility requirements with Queryverse [612083be] to versions: 0.20.0-0.21.8, leaving only versions: 0.21.0-0.21.8
 │   │         └─Queryverse [612083be] log:
 │   │           ├─possible versions are: 0.1.0-0.6.2 or uninstalled
 │   │           ├─restricted to versions * by an explicit requirement, leaving only versions 0.1.0-0.6.2
 │   │           └─restricted by compatibility requirements with DataFrames [a93c6f00] to versions: 0.6.1-0.6.2 or uninstalled, leaving only versions: 0.6.1-0.6.2
 │   │             └─DataFrames [a93c6f00] log: see above
 │   ├─restricted by compatibility requirements with TextParse [e0df1984] to versions: 0.6.0-0.7.0
 │   │ └─TextParse [e0df1984] log:
 │   │   ├─possible versions are: 0.6.0-1.0.1 or uninstalled
 │   │   ├─restricted by compatibility requirements with VegaDatasets [0ae4a718] to versions: 0.6.0-1.0.1
 │   │   │ └─VegaDatasets [0ae4a718] log:
 │   │   │   ├─possible versions are: 0.4.0-2.1.1 or uninstalled
 │   │   │   ├─restricted to versions * by an explicit requirement, leaving only versions 0.4.0-2.1.1
 │   │   │   └─restricted by compatibility requirements with DataStructures [864edb3b] to versions: 2.1.1 or uninstalled, leaving only versions: 2.1.1
 │   │   │     └─DataStructures [864edb3b] log: see above
 │   │   └─restricted by compatibility requirements with DataStructures [864edb3b] to versions: 1.0.1 or uninstalled, leaving only versions: 1.0.1
 │   │     └─DataStructures [864edb3b] log: see above
 │   └─restricted by compatibility requirements with CSVFiles [5d742f6a] to versions: 0.5.2-0.6.0, leaving only versions: 0.6.0
 │     └─CSVFiles [5d742f6a] log:
 │       ├─possible versions are: 0.9.0-1.0.0 or uninstalled
 │       └─restricted by compatibility requirements with Queryverse [612083be] to versions: 1.0.0
 │         └─Queryverse [612083be] log: see above
 ├─restricted by compatibility requirements with AbstractTrees [1520ce14] to versions: [0.4.1, 0.10.1-0.12.0] or uninstalled, leaving only versions: 0.10.1-0.10.4
 │ └─AbstractTrees [1520ce14] log:
 │   ├─possible versions are: 0.2.0-0.3.4 or uninstalled
 │   └─restricted by compatibility requirements with GraphRecipes [bd48cda9] to versions: 0.3.1-0.3.4
 │     └─GraphRecipes [bd48cda9] log:
 │       ├─possible versions are: 0.4.0-0.5.5 or uninstalled
 │       ├─restricted to versions 0.5 by Agents [46ada45e], leaving only versions 0.5.0-0.5.5
 │       │ └─Agents [46ada45e] log: see above
 │       └─restricted by compatibility requirements with RecipesBase [3cdcf5f2] to versions: 0.5.2-0.5.5 or uninstalled, leaving only versions: 0.5.2-0.5.5
 │         └─RecipesBase [3cdcf5f2] log:
 │           ├─possible versions are: 0.4.0-1.1.1 or uninstalled
 │           └─restricted to versions 1 by Agents [46ada45e], leaving only versions 1.0.0-1.1.1
 │             └─Agents [46ada45e] log: see above
 └─restricted by compatibility requirements with CuArrays [3a865a2d] to versions: [0.4.1-0.8.3, 0.11.0-0.12.0] or uninstalled — no versions left
   └─CuArrays [3a865a2d] log:
     ├─possible versions are: 0.2.1-2.2.2 or uninstalled
     ├─restricted by compatibility requirements with DataStructures [864edb3b] to versions: 0.2.1-1.2.1 or uninstalled
     │ └─DataStructures [864edb3b] log: see above
     └─restricted by julia compatibility requirements to versions: uninstalled
```

**Comment by Datseris** [comment]
Flux and other machine learning packages have not updated their `Require` versions to 1+ version. In Agents 4.2 we will have more loose `Requires` restrictions, but you will have to wait a couple of weeks for that. 

For now, please just use a new environment for Agents, or convince the authors of Flux to update the Requires version, or use CompatHelper so that it happens for them automatically.
